CMakeLists.txt

```diff
@@ -24,6 +24,12 @@ set_directory_properties(PROPERTIES EP_BASE "${PROJECT_BINARY_DIR}/external")
 
 add_definitions(-D_USE_MATH_DEFINES)
 
+# Check for OS
+if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+    set(MACOSX TRUE)
```

Comment:
A quick note for the future (if we decide to add FreeBSD support):

![Table from gitlab.kitware.com: CMAKE_SYSTEM_NAME is set to Darwin on (seemingly) only MacOS/iOS](https://user-images.githubusercontent.com/46334387/140759693-864bcc32-cb7f-49ef-8f5c-1836d128a3ae.png)

https://gitlab.kitware.com/cmake/community/-/wikis/doc/tutorials/How-To-Write-Platform-Checks#cmake-system

It looks like this won't be an issue if we decide to add FreeBSD support &mdash; `${CMAKE_SYSTEM_NAME}` is `FreeBSD` on FreeBSD.